### PR TITLE
(PC-24682) ci(yarn): POC using gsc for action cache

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-24682

## Checklist

I have:

- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]

## Screenshots

**delete** _if no UI change_

| Platform         | Mockup/Before | After |
| :--------------- | :-----------: | :---: |
| iOS              |               |       |
| Android          |               |       |
| Phone - Chrome   |               |       |
| Desktop - Chrome |               |       |

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

**Comment by lgerard-pass** [comment]
Tu as réussi à donner accès au service account du coup ? Tu as fait ça comment ?

File: .github/workflows/dev_on_workflow_install.yml [hunk 1]
```diff
@@ -18,16 +26,40 @@ jobs:
...
+      - id: cache-key
+        name: Create cache key
+        run: echo "cache_key=v1-yarn-dependency-cache-${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}" >> $GITHUB_OUTPUT
+      - id: has-cache
+        name: Check if cache exist
+        run: gsutil ls gs://${{ env.GCP_BUCKET_ID }}/${{ steps.cache-key.outputs.cache_key }}.tgz
+        continue-on-error: true
+      - name: Restore cache if existing
+        uses: mango-chutney/mango-actions-google-cache@releases/v1
+        if: ${{ steps.has-cache.outcome == 'success' }}
```
Comment (mlachevre-pass) [inline .github/workflows/dev_on_workflow_install.yml:47]: si tu continues on error, la tache est pas tjs en success ?

Comment (kopax-polyconseil) [reply .github/workflows/dev_on_workflow_install.yml:47]: `outcome` != `conclusion`

**Si succès** 

Demo: https://github.com/pass-culture/pass-culture-app-native/actions/runs/6301187743/job/17105699222#step:16:106

image 

**Si échec** 

Demo : https://github.com/pass-culture/pass-culture-app-native/actions/runs/6301375738/job/17106272306#step:16:106

image

**Comment by kopax-polyconseil** [comment]
> Tu as réussi à donner accès au service account du coup ? Tu as fait ça comment ?

Il fallait que la step `uses: 'google-github-actions/setup-gcloud@v1.1.0'` , soit après `uses: 'google-github-actions/auth@v1'`.

**Review by mlachevre-pass — APPROVED** [review]

**Pull request merged**